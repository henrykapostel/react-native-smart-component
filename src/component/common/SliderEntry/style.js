import { StyleSheet, Platform } from 'react-native'
import { ITEM_WIDTH, SLIDE_HEIGHT, ITEM_HORIZONTAL_MARGIN } from './../../../constants'

const entryBorderRadius = 8

export const styles = StyleSheet.create({
  imageContainer: {
    flex           : 1,
    backgroundColor: 'white',
    borderRadius   : entryBorderRadius
  },
  image: {
    ...StyleSheet.absoluteFillObject,
    resizeMode  : 'cover',
    borderRadius: Platform.OS === 'ios' ? entryBorderRadius : 0,
    flex        : 1,
  },
  slideInnerContainer: {
    width            : ITEM_WIDTH,
    height           : SLIDE_HEIGHT,
    paddingHorizontal: ITEM_HORIZONTAL_MARGIN,
    paddingBottom    : 18
  },
  radiusMask: {
    position: 'absolute',
    bottom  : 0,
    left    : 0,
    right   : 0,
    height  : entryBorderRadius
  },
})