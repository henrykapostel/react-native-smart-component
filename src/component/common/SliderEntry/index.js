import React, { Component } from 'react'
import { View, Image, TouchableOpacity } from 'react-native'
import { ParallaxImage } from 'react-native-snap-carousel'
import { styles } from './style'

export class SliderEntry extends Component {
  
  constructor(props) {
    
    super(props)
    this.state = {
      data         : props.data,
      parallax     : props.parallax,
      parallaxProps: props.parallaxProps,
      even         : props.even
    }
  }
  
  onShowAlert = () => {
    
    const { data: { objectId } } = this.state
    alert(`You've clicked '${objectId}'`)
  }
  renderImage () {
    
    const { data: { thumbnail }, parallax, parallaxProps } = this.state
    return parallax ? (
      <ParallaxImage
        source={{ uri: thumbnail }}
        containerStyle={styles.imageContainer}
        style={[styles.image, { position: 'relative' }]}
        parallaxFactor={0.35}
        showSpinner={true}
        spinnerColor="red"
        {...parallaxProps}
      />
    ) : (
      <Image
        source={{ uri: thumbnail }}
        style={styles.image}
      />
    )
  }
  
  render () {
    
    return (
      <TouchableOpacity
        activeOpacity={1}
        style={styles.slideInnerContainer}
        onPress={this.onShowAlert}
      >
        <View style={styles.imageContainer}>
          {this.renderImage()}
          <View style={styles.radiusMask} />
        </View>
      </TouchableOpacity>
    )
  }
}
