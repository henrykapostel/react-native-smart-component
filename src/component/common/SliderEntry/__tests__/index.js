import 'react-native'
import React from 'react'
import { SliderEntry } from './../../SliderEntry'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <SliderEntry data={{ thumbnail: '' }}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
