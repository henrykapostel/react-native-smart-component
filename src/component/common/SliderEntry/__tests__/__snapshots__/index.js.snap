// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 533.6,
      "opacity": 1,
      "paddingBottom": 18,
      "paddingHorizontal": 15,
      "width": 592.5,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 8,
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "borderRadius": 8,
          "bottom": 0,
          "flex": 1,
          "left": 0,
          "position": "absolute",
          "resizeMode": "cover",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      style={
        Object {
          "bottom": 0,
          "height": 8,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    />
  </View>
</View>
`;
